### PR TITLE
Upgrading Spark->3.3,Hadoop->3,Scala->2.13,Java->17

### PR DESCRIPTION
**Pull request by Bidek56**
## Describe your changes
Upgrading:
  - Spark -> 3.3
  - Hadoop -> 3.3
  - Scala -> 2.13
  - Java -> 17

Breaking change: [spylon-kernel ](https://github.com/vericast/spylon-kernel) not longer works since the package is no longer maintained. 
For anyone looking for a Scala kernel, I would suggest looking at [Almond](https://github.com/almond-sh/almond)

## Checklist (especially for first-time contributors)

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests
- [x] I will try not to use force-push to make the review process easier for reviewers
- [x] I have updated the documentation for significant changes

### CODE REVIEW

**Comment by Bidek56** [comment]
@mathbunnyru Let me know what you think we should do with spylon-kernel which is causing tests to fail.
Because spylon-kernel is a dead package, I think we should remove it and update our documentation to use Almond instead.
Thanks

**Comment by bjornjorgensen** [comment]
Almound dont support spark 3.x https://github.com/almond-sh/almond/issues/929

**Comment by bjornjorgensen** [comment]
@Bidek56 [her](https://github.com/jupyter/docker-stacks/compare/master...bjornjorgensen:master) I made some changes to this PR

File: tests/all-spark-notebook/test_spark_notebooks.py
Comment (mathbunnyru) [inline tests/all-spark-notebook/test_spark_notebooks.py:42]: Just remove `no_warnings=False,` line and all lines about the warnings.

**Review by mathbunnyru — CHANGES_REQUESTED** [review]
I would suggest doing it this way:

1. Remove scala in a separate PR first. We can't keep using software which is not updated and starts breaking things. If possible, it would be nice to have a replacement, like `almond` or sth else (when `almond` or something similar is working in our environment).

2. Then we can merge this PR.

**Comment by bjornjorgensen** [comment]
@Bidek56 Will you do this? If not than I can do it. If you will try I can help you, just ping me.. 
@mathbunnyru Nice plan. I havent found any scala kernel that works with spark 3.X.

**Comment by bjornjorgensen** [comment]
I have created a [branch](https://github.com/bjornjorgensen/docker-stacks/tree/Spark-3.3.0,Hadoop-3,Scala-2.13,Java-17) for this upgrade just to test it. 

[docker imgae for pyspark](https://hub.docker.com/repository/docker/bjornjorgensen/docker-stack-spark3.3.0) 
and 
[docker imgae for all-spark-notebook](https://hub.docker.com/repository/docker/bjornjorgensen/docker-stack-allspark3.3.0) 

I test the scala kernel and only get `Intitializing Scala interpreter ...`

File: tests/all-spark-notebook/test_spark_notebooks.py
Comment (Bidek56) [reply tests/all-spark-notebook/test_spark_notebooks.py]: Done.

**Comment by Bidek56** [comment]
Just a note that this PR will break [Build an Image with a Different Version of Spark](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/specifics.html#build-an-image-with-a-different-version-of-spark) for Spark < 3.2 since [Apache](https://archive.apache.org/dist/spark/) was not providing tgz files with Scala version appended.

**Review by mathbunnyru — APPROVED** [review]
@Bidek56 as you mentioned, could you please update the docs, so they will work after this change is merged?

**Comment by mathbunnyru** [comment]
I merged removing `spylon-kernel`. I will have to do a few full rebuilds, to make sure it's been removed in all images.
Please, merge master to this branch as well.

**Comment by Bidek56** [comment]
The build has failed because it is running Spark 3.2.1 not 3.3. 
Anyone knows where it's getting the wrong version of Spark from? Thx

`file:/usr/local/spark-3.2.1-bin-hadoop3.2/jars/spark-unsafe_2.12-3.2.1.jar`

**Comment by mathbunnyru** [comment]
@Bidek56 this happens because when building `all-spark-notebook` instead of using freshly build `pyspark-notebook` we use old image which we pull.
This is true for other images as well and it's my nightmare 😂 
I've just finished my work to completely rewrite CI system from scratch.
Please, take a look here:
https://github.com/jupyter/docker-stacks/pull/1703

After I merge this PR, I think we will be able to easily merge your PR.

**Comment by bjornjorgensen** [comment]
Yes, you don't have to do anything with the tests. 
And you can remove https://github.com/jupyter/docker-stacks/blob/61b20aa69ed8c726194a9865232ef71924fb47d3/pyspark-notebook/Dockerfile#L51-L55

**Comment by mathbunnyru** [comment]
@Bidek56 could you merge master once again?
I merged my PR, so theoretically, everything should work fine in yours.

File: docs/using/specifics.md
Comment (mathbunnyru) [inline docs/using/specifics.md:50]: This sentence is not clear to me. Could you take a look please?

File: docs/using/specifics.md
Comment (mathbunnyru) [inline docs/using/specifics.md:44]:
```suggestion
- Spark distribution is defined by the combination of Spark, Hadoop and Scala versions and verified by the package checksum,
```

File: docs/using/specifics.md
Comment (mathbunnyru) [inline docs/using/specifics.md:55]:
```suggestion
- Starting with _Spark >= 3.2_ the distribution file contains Scala version, hence building older Spark will not work.
- Building older version requires modification to the Dockerfile or using it's older version
```

**Review by mathbunnyru — COMMENTED** [review]
Some minor documentation comments only.

File: docs/using/specifics.md
Comment (bjornjorgensen) [reply docs/using/specifics.md:55]: Can we change _Spark >= 3.2_ to Spark >= 3.2?

Comment (mathbunnyru) [reply docs/using/specifics.md:55]: I'm ok with both options :)

Comment (bjornjorgensen) [reply docs/using/specifics.md:55]: Oh.. its styling.. hehe I did't know :)

Comment (mathbunnyru) [reply docs/using/specifics.md:55]: Using italic just makes it clear that `>= 3.2` is about spark version.

File: docs/using/specifics.md
Comment (mathbunnyru) [reply docs/using/specifics.md]: Great, thank you! 👍

**Review by mathbunnyru — APPROVED** [review]
I will merge this one when tests pass.
Thank you for your contribution @Bidek56 👍

**Comment by alexander-manley** [comment]
All checks have passed. Excellent!
33 successful checks

**Pull request merged**
